### PR TITLE
Add Tooltip documentation page

### PR DESCRIPTION
**Pull request by farazcsk**

### CODE REVIEW

File: styleguide/screens/Patterns/Tooltip/Tooltip.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+            >
+              <Content />
+            </Tooltip>
+          </Specimen>
+        </D>
+        <D>
+          <H level={ 2 } className={ m.mtr }>Positioning</H>
+          <T elm="p" className={ m.mtr }>
+            Tooltips are based on
+            the <Link className={ scaffoldCss.link } to="/patterns/tether/">Tether</Link> component
```
Comment (rdjpalmer) [inline styleguide/screens/Patterns/Tooltip/Tooltip.js]: We should probably document Tether before merging this in

File: styleguide/screens/Patterns/Tooltip/Tooltip.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+              <Tooltip
+                target={ <DefaultTarget /> }
+                verticalAttachment={ VERTICAL_ATTACHMENTS.TOP }
+                horizontalAttachment={ HORIZONTAL_ATTACHMENTS.CENTER }
+                active
+              >
+                <Content />
+              </Tooltip>
+            ` }
+            variant="dark"
```
Comment (rdjpalmer) [inline styleguide/screens/Patterns/Tooltip/Tooltip.js]: This doesn't look so great with the lighter modal on over the top of it. I'd say we should keep it light, despite it kind of breaking the rules defined in the documentation

File: styleguide/screens/Patterns/Tooltip/Tooltip.js [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+          >
+            <Tooltip
+              target={
+                <LightTarget
+                  onMouseEnter={ this.handleMouseEnter }
+                  onMouseLeave={ this.handleMouseLeave }
+                />
+              }
+              verticalAttachment={ VERTICAL_ATTACHMENTS.CENTER }
+              horizontalAttachment={ HORIZONTAL_ATTACHMENTS.RIGHT }
```
Comment (rdjpalmer) [inline styleguide/screens/Patterns/Tooltip/Tooltip.js]: We don't need to show different attachments as part of this documentation—the link top the Tether documentation will suffice

File: styleguide/screens/Patterns/Tooltip/Tooltip.js [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+  <button
+    onMouseEnter={ () => { onMouseEnter('showDefaultTooltip'); } }
+    onMouseLeave={ () => { onMouseLeave('showDefaultTooltip'); } }
+  >
+    Hover on me
+  </button>
+);
+
+// eslint-disable-next-line react/prop-types
+const LightTarget = ({ onMouseEnter, onMouseLeave }) => (
```
Comment (rdjpalmer) [inline styleguide/screens/Patterns/Tooltip/Tooltip.js]: The target should probably be consistent with other pages and be one of the `Placeholder` components found in `Scaffold.js`, [as per the modal documentation](https://bloom.appearhere.co.uk/patterns/modals).

I'd prefer for the tooltip to show on click too, as that's the best event to use for cross device support. I'd be tempted to remove any mention of `hover` from the documentation too, but that's open for debate

File: styleguide/screens/Patterns/Tooltip/Tooltip.js [hunk 5]
```diff
@@ -13,42 +13,38 @@ import css from './Tooltip.css';
 import scaffoldCss from '../../../components/Scaffold/Scaffold.css';
 
 // eslint-disable-next-line react/prop-types
-const DefaultTarget = ({ onMouseEnter, onMouseLeave }) => (
-  <button
-    onMouseEnter={ () => { onMouseEnter('showDefaultTooltip'); } }
-    onMouseLeave={ () => { onMouseLeave('showDefaultTooltip'); } }
+const DefaultTarget = ({ onClick }) => (
+  <Placeholder
+    onClick={ () => { onClick('showDefaultTooltip'); } }
```
Comment (rdjpalmer) [inline styleguide/screens/Patterns/Tooltip/Tooltip.js]: No need to multiline this as it has one prop

File: styleguide/screens/Patterns/Tooltip/Tooltip.js [hunk 6]
```diff
@@ -74,13 +70,11 @@ export default class TooltipDocumentation extends Component {
           <Specimen
             classNames={ {
               root: m.mtr,
-              specimenContainer: cx(m.par, css.specimenContainer),
+              specimenContainer: cx(css.specimenContainer),
```
Comment (rdjpalmer) [inline styleguide/screens/Patterns/Tooltip/Tooltip.js]: No need for `cx` here

File: styleguide/screens/Patterns/Tooltip/Tooltip.js (hunk 4 above)
Comment (rdjpalmer) [reply styleguide/screens/Patterns/Tooltip/Tooltip.js]: @farazcsk what are your thoughts on the following?

> I'd be tempted to remove any mention of hover from the documentation too, but that's open for debate

Comment (farazcsk) [reply styleguide/screens/Patterns/Tooltip/Tooltip.js]: @rdjpalmer So I actually already made the changes stated above as I totally agree with point made in the documentation about not using tooltips on mobile, but In hindsight I think that we should remove the mention of click 😆 .

I'm probably wrong, but I don't think tooltips were ever designed to be shown on click and mentioning that they basically won't be seen on mobile devices is probably enough.

**Comment by farazcsk** [comment]
So after some thought, I don't think documenting this component is useful as it is currently not in use at @appearhere. For now the documentation will be parked, at least until we have a genuine use case for this component which we can use as the basis for our it's documentation.

**Pull request closed without merging**